# SF-Coyote-Diet-Metabarcoding

**Title:** Urban landscape heterogeneity shapes individual and territory-level variation in coyote diet

**Abstract:** In the past decade, studies have demonstrated that several traits, including foraging behavior and diet, differ between urban and nonurban wildlife populations. However, little is known about how environmental heterogeneity shapes dietary variation of organisms within cities. We examined the diets of coyotes (*Canis latrans*) in San Francisco to quantify territory- and individual-level dietary differences and determine how within-city variation in land cover and land use affect coyote diet. We genotyped fecal samples for individual coyote identification and quantified diet with DNA metabarcoding. The highest contributor to coyote diet was anthropogenic food followed by small mammals. The most frequently detected species were domestic chicken, pocket gopher (*Thomomys bottae*), pig, and raccoon (*Procyon lotor*). Diet composition varied significantly across territories and among individuals. Within family groups, however, individual diets were relatively consistent. The representation of anthropogenic food in scats was correlated with impervious surface cover, suggesting that coyotes consumed more human food in more urbanized territories. The representation of invasive, human-commensal rodents in the diet was correlated with the number of food services in a territory. Overall, our results revealed substantial intraspecific variation in coyote diet associated with urban heterogeneity and point to a diversifying effect of urbanization on population diet.

**Code version:** R v 4.2.1

**To add:**

-   Overview of folders/files and their contents

-   Instructions for users to run the software (e.g. explain the project workflow and any configuration parameters of your software)

-   Links to protocols.io or equivalent methods repositories, where applicable

-   Once a paper is accepted, please include author names, contact details, links to preprint and the publication.

## Data Files:

### Filtering-and-QC

-   batch1_reads.csv and batch2_reads.csv: ASV tables generated by DADA2 with taxonomy assigned via BLAST added.
-   All_Metadata.csv: metadata for all fecal samples
-   NoCanis_Scats.csv: list of samples names that did not generate Canis spp. reads
-   Species_FunctionalGroups.csv: categorizes diet items into functional groups

### Diet-Plots

-   12S_species_categories.csv: categorizes diet items into taxonomic groups

-   IDs.csv: lists sex and family group assignment of individual coyotes

### Regression-Analyses

-   Territory_All_Covariates.csv: land cover and land use covariates for each territory

## Code:

## 1. Sequence-Processing

-   Trimming-Reads.Rmd: trim fastq files with cutadapt. High performance computing required.
-   DADA2.Rmd: identify ASVs with DADA2. High performance computing required.

## 2. Filtering-and-QC

-   FilteringReads.Rmd: R scripts for filtering denoised data. The input files are ASV tables output by DADA2 with taxonomy assigned by BLAST+.

## 3. Data Visualization and Statistical Analyses

Creating figures or conducting statistical analyses requires that filtering and QC steps have been run as filtered and finalized data frames are needed for visualizing and analyzing diet data.

### Diet-Plots

-   Code to generate figures 1 and 2 in the main text

### iNEXT: Sample Coverage, Species Richness, Species Diversity

-   Script for generating rarefaction curve plots and calculating diversity metrics for both territories and individuals

### Ordination (nMDS)

-   Add details here

### PERMANOVA

-   Add details here

### Regression Analyses

-   Add details here for statistical analyses
-   Code to generate figure 3 in the main text

### SIMPER

-   Add details here

### Replication Analyses

-   Add details here
-   30 extraction replicates that were actually metabarcoded; 44 PCR replicates from mini exp 7 that intersect; 78 PCR replicates from mini exp 1 that intersect; In the end, 10+54+23 = 87 pairs remaining

### Figure for Controls
